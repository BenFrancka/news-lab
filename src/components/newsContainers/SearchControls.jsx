import React from 'react';
import PropTypes from 'prop-types';

const SearchControls = ({ searchTerm, onSearchChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <label htmlFor="searchTerm">Search Term</label>
    <input
      id="searchTerm"
      type="text"
      value={searchTerm}
      onChange={onSearchChange}
    />
    <button aria-label="search-articles">Search</button>
  </form>
);

SearchControls.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  onSearchChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchControls;

