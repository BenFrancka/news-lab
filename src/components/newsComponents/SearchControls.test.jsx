import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchControls from './SearchControls';

describe('SearchControls', () => {
  it('should be comprised on an input and a button', async () => {
    render(<SearchControls />);

    const input = await screen.findByLabelText('Search Term');
    userEvent.type(input, 'snakes');

    const submitButton = await screen.findByRole('button', {
      name: 'search-articles',
    });
    
    expect(input).toMatchSnapshot();
    expect(submitButton).toMatchSnapshot();
  });
});
