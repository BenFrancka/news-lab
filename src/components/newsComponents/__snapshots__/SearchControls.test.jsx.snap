// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchControls should be comprised on an input and a button 1`] = `
<input
  id="searchTerm"
  type="text"
  value=""
/>
`;

exports[`SearchControls should be comprised on an input and a button 2`] = `
<button
  aria-label="search-articles"
>
  Search
</button>
`;
